lib/logstash/filters/aggregate.rb

```diff
@@ -384,10 +433,31 @@ def remove_expired_maps()
     min_timestamp = Time.now - @timeout
     
     @@mutex.synchronize do
+
       @@aggregate_maps.delete_if do |key, element| 
         if (element.creation_timestamp < min_timestamp)
-          if (@push_previous_map_as_event)
-            events_to_flush << LogStash::Event.new(element.map)
+          if (@push_previous_map_as_event) || (@push_map_as_event_on_timeout)
+            event = LogStash::Event.new(element.map)        
+            
+            if @timeout_task_id_field
+              event_to_yield[@timeout_task_id_field] = task_id
+            end
+
+            # Call code block if available
+            if @timeout_code
+              begin
+                @timeout_codeblock.call(event)
+                events_to_flush << event
+                noError = true
+              rescue => exception
+                @logger.error("Aggregate exception occurred. Error: #{exception} ; Code: #{@timeout_code} ; EventData: #{event.instance_variable_get('@data')}")
+                event.tag("_aggregateexception")
+              end
+            else
+              events_to_flush << event
+            end
+
```

Comment:
Up to me, code `events_to_flush << event` must be present only one time just after this `end` so that generated event if always flushed, even if `timeout_code` execution failed.